File: widgetssdk/src/main/java/com/glia/widgets/chat/domain/CustomCardAdapterTypeUseCase.java
package com.glia.widgets.chat.domain;

import androidx.annotation.Nullable;

import com.glia.androidsdk.chat.ChatMessage;
import com.glia.widgets.chat.adapter.CustomCardAdapter;

public class CustomCardAdapterTypeUseCase {
    @Nullable
    private final CustomCardAdapter adapter;

    public CustomCardAdapterTypeUseCase(@Nullable CustomCardAdapter adapter) {
        this.adapter = adapter;
    }

    @Nullable
    public Integer execute(ChatMessage message) {
        if (adapter == null || message.getMetadata() == null || message.getMetadata().length() == 0) {
            return null;
        }
        return adapter.getChatAdapterViewType(message);
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/chat/domain/CustomCardAdapterTypeUseCase.kt
package com.glia.widgets.chat.domain

import com.glia.androidsdk.chat.ChatMessage
import com.glia.widgets.chat.adapter.CustomCardAdapter

class CustomCardAdapterTypeUseCase(private val adapter: CustomCardAdapter?) {
    operator fun invoke(message: ChatMessage): Int? = when {
        adapter == null || message.metadata == null || message.metadata!!.length() == 0 -> null
        else -> adapter.getChatAdapterViewType(message)
    }
}
